### PR TITLE
CLC-6397 OEC: Enforce matching header rows when parsing from CSV

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.ets.berkeley.edu/jira/browse/CLC-6397

The code wanted to error out on mismatch, but didn't quite know how. Spec for reassurance.

### CODE REVIEW

**Comment by johncrossman** [comment]
Travis has got dem blues. Err, I mean reds.

File: app/models/oec/worksheet.rb [hunk 1]
```diff
@@ -38,7 +38,7 @@ def self.from_csv(csv, opts={})
       return unless csv && (parsed_csv = CSV.parse csv)
       instance = self.new opts
       (header_row = parsed_csv.shift) until (header_row == instance.headers || parsed_csv.empty?)
-      raise ArgumentError, "Header mismatch: cannot create instance of #{self.name} from CSV" unless header_row
+      raise ArgumentError, "Header mismatch: cannot create instance of #{self.name} from CSV" unless (header_row == instance.headers)
```
Comment (johncrossman) [inline app/models/oec/worksheet.rb]: @paulkerschen would a more informative message be useful to our customer? Eg, "expected X; received Y"

Comment (paulkerschen) [inline app/models/oec/worksheet.rb]: Makes sense, @johncrossman. Probably the safest thing to log is the unparsed CSV text, limiting to maybe 500 characters - and I'll see about those, er, 48 Travis failures.

**Comment by paulkerschen** [comment]
Thanks for review, @johncrossman. PR amended.

**Comment by johncrossman** [comment]
Looks good. All eyes on Travis...

**Pull request merged**